tools/clang/lib/SPIRV/PervertexInputVisitor.cpp

```diff
@@ -310,8 +317,28 @@ bool PervertexInputVisitor::visit(SpirvFunctionCall *inst) {
   unsigned argIndex = 0;
   for (auto *arg : inst->getArgs()) {
     if (currentFunc->getMappedFuncParam(arg)) {
-      createVertexStore(arg,
-                        createVertexLoad(currentFunc->getMappedFuncParam(arg)));
+      auto paramVar = currentFunc->getMappedFuncParam(arg);
+      if (isa<SpirvAccessChain>(paramVar)) {
+        auto tempVar = paramVar;
+        while (isa<SpirvAccessChain>(tempVar)) {
+          tempVar = dyn_cast<SpirvAccessChain>(tempVar)->getBase();
+        }
+        if (tempVar->isNoninterpolated()) {
+          /// For Structure type, expanded param needs restore to
```

Comment:
I find the wording in the comment confusing. As someone not too familiar, I do not understand what the means.

Do you mean something like:

> If the base variable is not interpolated, then we need to copy the values for all of the vertices to the local variable. This means copying an entire array. This is accomplished by removing the last index in the access chain for the parameter variable to get a pointer to the entire array.

Reply:
Yes. A minor difference is we need to recreate (clone) a new access chain instruction instead of just remove original last zero index, I'll show an example in following comments.